socorro/processor/rules/mozilla.py

```diff
@@ -626,10 +626,57 @@ def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
                 return
 
 
+class MissingSymbolsRule(Rule):
+    """
+    Adds ``missing_symbols`` field where the value is a semi-colon separated set of
+    ``module/version/debugid`` strings for modules where the stackwalker couldn't find a
+    symbols file.
+    """
+
+    # Filenames should contain A-Za-z0-9_.- and that's it.
+    BAD_FILENAME_CHARACTERS = re.compile(r"[^a-zA-Z0-9_\.-]", re.IGNORECASE)
+
+    # Debug ids are hex strings
+    BAD_DEBUGID_CHARACTERS = re.compile(r"[^a-f0-9]", re.IGNORECASE)
+
+    NULL_DEBUG_ID = "0" * 33
+
+    def format_module(self, item):
+        filename = item["filename"]
+        filename = self.BAD_FILENAME_CHARACTERS.sub("", filename)
+
+        version = item.get("version", "") or "None"
+        version = version.replace("/", "\\/")
+
+        debugid = item.get("debug_id", self.NULL_DEBUG_ID)
+        debugid = self.BAD_DEBUGID_CHARACTERS.sub("", debugid)
```

Comment:
I've seen wild filenames, so normalizing those to valid values is important. I haven't seen wild versions or wild debug ids, yet, so the validation here is half-hearted.